words/scanner.go

```diff
@@ -143,6 +171,9 @@ func SplitFunc(data []byte, atEOF bool) (advance int, token []byte, err error) {
 					w = w2
 
 					pos += w
```

Comment:
Is it possible in all of these different cases that `pos += w` actually takes us over the limit?

Reply:
Yes that's possible, going to see how Solr handles that

Reply:
Solr doesn't care about the byte size, so I'll change this to increment a new var instead